fabtests/common/shared.c

```diff
@@ -3216,6 +3217,37 @@ void ft_parse_opts_range(char* optarg)
 	test_size = range_test_size;
 }
 
+void ft_parse_opts_list(char* optarg)
+{
+	int i, ret;
+	char *token;
+
+	optarg += 2; // remove 'l:'
+	test_cnt = 1;
+	for (i=0; optarg[i] != '\0'; i++) {
+		test_cnt += optarg[i] == ',';
+	}
+	range_test_size = calloc(test_cnt, sizeof(*range_test_size));
+	if (!range_test_size) {
+		perror("calloc");
+		exit(EXIT_FAILURE);
+	}
+
+	token = strtok(optarg, ",");
+	i = 0;
+	while (token != NULL) {
+		range_test_size[i].enable_flags = 0;
+		ret = sscanf(token, "%ld", &range_test_size[i].size);
+		if (ret != 1) {
+			perror("sscanf cannot parse integer in list");
+			exit(EXIT_FAILURE);
+		}
+		i++;
```

Comment:
If we use test_cnt instead of i in this second loop, extra ```,``` should be silently discarded.